beacon-chain/stategen/service.go

```diff
@@ -16,3 +22,21 @@ func New(db db.NoHeadAccessDatabase) *State {
 		beaconDB: db,
 	}
 }
+
+// GenerateState generates a state from the provided pre-state till the provided block.
+func (s *State) GenerateState(ctx context.Context, preState *stateTrie.BeaconState, endBlock *ethpb.SignedBeaconBlock) (*stateTrie.BeaconState, error) {
```

Comment:
can you put this back in block chain `service.go`, you can combine it with: https://github.com/prysmaticlabs/prysm/pull/4844/files#diff-c2ffc98dd863a51dc11c4843367b82b6R160

this API pattern with `preState` and `endBlock` seem to be only serving block chain service specific use case, because of that, we shouldn't put it here.

I'll export `loadBlocks` and `replayBlocks` via an interface for you

Reply:
Wouldn't something like this also be useful for other services, ex: Sync,etc ? You can generate a new state from a provided pre-state

Reply:
I agree, but we don't know what the API is for that yet. I think until sync actually needs it / uses it. It's better to constraint this within the block chains service. 
Or else it may introduce disparity where chain service uses one API to generate state and sync service uses another API to generate state and both APIs reside in stategen service